dask/dataframe/core.py

```diff
@@ -108,7 +108,7 @@ def finalize(results):
 class Scalar(DaskMethodsMixin, OperatorMethodMixin):
     """ A Dask object to represent a pandas scalar"""
 
-    def __init__(self, dsk, name, meta, divisions=None):
+    def __init__(self, dsk, name, meta, divisions=None, partition_sizes=None):
```

Comment:
The sizes of partitions are now tracked so that we can effectively align/zip long data structures for processing, and do it correctly.   In some cases there is work to make partitions align by breaking both up, but in our common cases, they happen to align already, since they are split on per-sample h5ad file boundaries.